WebApplicationDemo/Application/TestContextAsync.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Ark.Tools.Outbox.SqlServer;
+using Ark.Tools.Sql;
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WebApplicationDemo.Application
+{
+    public class TestContextAsync : SqlContextAsyncFactory<ITestContext>
+    {
+        public TestContextAsync(IDataContextConfig contextConfig, IDbConnectionManager dbConnectionManager) : 
+            base(dbConnectionManager, contextConfig.SqlConnectionString)
+        {
+        }
+
+        public override ITestContext Create(DbConnection dbConnection, DbTransaction dbTransaction)
+        {
+            var ctx = new TestContext(dbConnection, dbTransaction);
+
+            return ctx;
+        }
+    }
+
+
+    public interface ITestContext : ISqlContextAsync, IAsyncDisposable
+    { 
+
+    }
+
+    public class TestContext : ITestContext
+    {
+
+        public DbConnection Connection { get; set; }
+
+        public DbTransaction? Transaction { get; set; }
+
+        public TestContext(DbConnection dbConnection, DbTransaction dbTransaction)
+        {
+            Connection = dbConnection;
+            Transaction = dbTransaction;
+        }
+
+        public async ValueTask ChangeIsolationLevelAsync(IsolationLevel isolationLevel, CancellationToken ctk)
+        {
+            Transaction = await Connection.BeginTransactionAsync(isolationLevel, ctk);
```

Comment:
this is not right as Transaction may be not null, and you're Leaking a Disposable resource.

Disposable shoukd be disposed before being replaced